homeassistant/helpers/selector.py

```diff
@@ -38,12 +49,27 @@ class Selector:
     """Base class for selectors."""
 
     CONFIG_SCHEMA: Callable
+    config: Any
+    name: str
+
+    def __init__(self, config: Any) -> None:
+        """Instantiate a selector."""
+        self.config = self.CONFIG_SCHEMA(config)
+
+    def serialize(self) -> Any:
+        """Serialize Selector for voluptuous_serialize."""
+        result: dict[str, Any] = {"selector": self.name}
+        if self.config:
+            result["config"] = self.config
+        return result
```

Comment:
```suggestion
        return {"selector": { self.name: self.config } }
```

https://github.com/home-assistant/frontend/blob/dev/src/components/ha-form/types.ts#L30